File: Python/Leetcode 443.py
class Solution:
    def compress(self, chars: List[str]) -> int:
        temp_list = []

        st = 0
        length = 0

        while st < len(chars):
            curr_st = st
            count = 0

            while st < len(chars) and chars[st] == chars[curr_st]:
                count = count + 1
                st = st + 1
            
            temp_list.append(chars[curr_st])
            length = length + 1

            if count > 1:
                temp_count = str(count)

                for ch in temp_count:
                    temp_list.append(ch)
                
                length = length + len(temp_count)
        
        for index in range(min(len(chars),len(temp_list))):
            chars[index] = temp_list[index]
        
        for index in range(len(chars),len(temp_list)):
            chars.append(temp[index])
        
        return length

File: C++/Leetcode 443.cpp
class Solution {
public:
    int compress(vector<char>& chars) {
        vector<char> temp;

        int st = 0;
        int length = 0;
        while (st < chars.size())
        {
            int curr_st = st;
            int count = 0;

            while (st < chars.size() && chars[st] - 'a' == chars[curr_st] - 'a')
            {
                count++;
                st++;
            }
            temp.push_back(chars[curr_st]);
            length++;
            if (count > 1)
            {
                string temp_count = to_string(count);

                for (int i = 0; i < temp_count.length(); i++)
                {
                    temp.push_back(temp_count[i]);
                }
                // temp.push_back(to_string(count));
                length += ((to_string(count).length()));
            }
        }

        for (int i = 0; i < temp.size(); i++)
        {
            cout << temp[i] << "  ";
        }

        // chars = vector<int>(temp.begin(),temp.end());
        for (int i = 0; i < min(chars.size(),temp.size()); i++)
        {
            chars[i] = temp[i];
        }

        for (int i = chars.size(); i < temp.size(); i++ )
        {
            chars.push_back(temp[i]);
        }
        return length;
    }
};